session/manager_instance.go

```diff
@@ -32,25 +32,26 @@ type IDGenerator func() SessionID
 // SaveCallback stores newly started sessions
 type SaveCallback func(Session)
 
-// NewManager returns session manager which maintains a map of session id -> session
-func NewManager(idGenerator IDGenerator, configProvider ServiceConfigProvider, saveCallback SaveCallback) *manager {
-	return &manager{
+// NewManager returns new session ManagerInstance
+func NewManager(idGenerator IDGenerator, configProvider ServiceConfigProvider, saveCallback SaveCallback) *ManagerInstance {
+	return &ManagerInstance{
 		generateID:     idGenerator,
 		generateConfig: configProvider,
 		saveSession:    saveCallback,
 		creationLock:   sync.Mutex{},
 	}
 }
 
-type manager struct {
+// ManagerInstance knows how to start and provision session
+type ManagerInstance struct {
```

Comment:
Not a fan of having "instance" in struct names, but I do see why you use it here

Reply:
If it's hard to name it - it means its hard to define purpose

Reply:
Ditched this renaming. Will work on better `SessionManager` implementation and `SessionHandler` interface later